rest/client/rest.go

```diff
@@ -551,3 +552,57 @@ func (c *communicatorImpl) GetClientConfig(ctx context.Context) (*evergreen.Clie
 
 	return &config, nil
 }
+
+func (c *communicatorImpl) GetSubscriptions(ctx context.Context) ([]event.Subscription, error) {
+	info := requestInfo{
+		path:    "/subscriptions",
+		method:  get,
+		version: apiVersion2,
+	}
+	resp, err := c.request(ctx, info, nil)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to fetch subscriptions")
+	}
+	if resp.StatusCode != http.StatusOK {
+		restErr := &rest.APIError{}
+		if err = util.ReadJSONInto(resp.Body, restErr); err != nil {
```

Comment:
This should be `err == nil`, since in this case we've successfully unmarshaled into `restErr`.

Reply:
I flipped the return statements around, so this logic is now correct